src/clj/qbits/checkmate.clj

```diff
@@ -99,3 +99,25 @@
 
     (attempt [_ f]
       (try-attempt f))))
+
+(def async-runner-ch
+  (reify RetryStrategy
+    (run [this f opts]
+      (let [{:keys [delays success error failure] :as opts}
+            (merge {:delays (take 100 (constant-backoff 100))
+                    :success identity}
+                   opts)]
+        (async/go-loop [delays delays]
+          (let [ret (async/<! (f))]
+            (cond
+              (instance? Exception ret)
```

Comment:
The idea is that f returns the exception instead of throwing it and breaking the flow. This behavior is at the base of how max uses channel in general, so I preserve this pattern and it fits into sink easily.